### PR TITLE
Refactor helper functions

### PR DESCRIPTION
**Pull request by C-likethis123**

### CODE REVIEW

File: src/utils/inferencerUtils.ts [hunk 1]
```diff
@@ -75,32 +75,48 @@ const predefined = new Set([
...
   console.log('Printing Type Environment')
   for (let [key, value] of typeEnvironment) {
     if (predefined.has(key)) {
       continue
     }
-    if (typeof value === 'object') {
-      value = JSON.stringify(value)
-    }
-    console.log(`${key} = T${value}`)
+    console.log(`${key} = ${printType(value.types[0])}`)
```
Comment (maylim17) [inline src/utils/inferencerUtils.ts:112]: `${printType(value.types[0])}`)` - I think this part will give error. 

The current TypeEnv is `...types: [{argumentTypes: [Type, Type], resultType: Type}]...` 
So `value.types[0]` is technically `[{argumentTypes: [Type, Type], resultType: Type}]` which isn't a Type object so `printType()` function won't work

I thought of your suggestion about changing the above to Type objects, but I realized it doesn't really make sense. E.g. like when we declare the type for BinaryExpression with operator `*` in TypeEnv, its types isn't related to a functionType although the syntax looks similar. Like I feel we shouldn't make it into something it isn't. So I think the current TypeEnv syntax should stay for the time being

File: src/utils/inferencerUtils.ts [hunk 2]
```diff
@@ -75,10 +75,30 @@ const predefined = new Set([
...
+      // remove last comma
+      params = params.replace(/,\s*$/, "");
+      const returnType = printType(type.returnType)
+      return `(${params}) => ${returnType}`
+    default:
+      return "Not included in Source 1!"
+  }
+}
+
+export function printTypeConstraints(typeContraints: Map<number, Type>) {
```
[the remark this answers is not in the document]
Comment (maylim17) [reply src/utils/inferencerUtils.ts]: I think this has been updated right?

File: src/utils/inferencerUtils.ts (hunk 1 above)
Comment (C-likethis123) [reply src/utils/inferencerUtils.ts:112]: If I'm not wrong, aren't binary and unary operators functions? They take in arguments and return a result.

Comment (maylim17) [reply src/utils/inferencerUtils.ts:112]: Hmm okay makes sense. Then I'll refractor the TypeEnv ba. In that case you can push urs, it should work fine after my refactor

**Pull request merged**